packages/wallets/src/components/WalletHistory.tsx

```diff
@@ -263,9 +265,10 @@ export default function WalletHistory(props: Props) {
         metadata={metadata}
         expandedCellShift={1}
         uniqueField="name"
-        onToggleExpand={async (rowId: string, expanded: boolean) => {
-          const memos = (await getTransactionMemo({ transactionId: rowId.split('-')[0] })).data;
-          if (expanded) {
+        onToggleExpand={async (rowId: string, wasExpanded: boolean) => {
+          if (wasExpanded) {
+            const memos = mapMemos[rowId] || (await getTransactionMemo({ transactionId: rowId.split('-')[0] })).data;
+            mapMemos[rowId] = memos;
             const newTransactions = transactions.map((transaction) =>
```

Comment:
Looks much better, thanks. If I'm being nitpicky, I would only change mapMemos[rowId] IF the memo was just fetched from getTransactionMemo. In other words, there's no need to update the dict if the memo came from there.

Reply:
I made a few additional changes in this area. Resolving this now.